IPs/RSKIP428.md

```diff
@@ -0,0 +1,75 @@
+---
+rskip: 428
+title: Emit a new event with the utxo outpoints when a peg-out is created
+created: 23-APR-24
+author: NC
+purpose: Sca
+layer: Core 
+complexity: 1
+status: Draft
+description: 
+---
+
+|RSKIP          | 428                                                                |
+| :------------ |:-------------------------------------------------------------------|
+|**Title**      | Emit a new event with the utxo outpoints when a peg-out is created |
+|**Created**    | 23-APR-24                                                          |
+|**Author**     | NC                                                                 |
+|**Purpose**    | Sca                                                                |
+|**Layer**      | Core                                                               |
+|**Complexity** | 1                                                                  |
+|**Status**     | Draft                                                              |
+
+## Abstract
+
+This RSKIP proposes creating a new event that stores the outpoint values of the inputs used in a peg-out to enable the PowHSM sign segwit peg-outs.
+
+## Motivation
+
+For the PowHSM sign segwit peg-outs[1], the outpoint values of the inputs must be provided since they are needed for calculating the sighash of a segwit bitcoin transaction. Currently, selected UTXOs for a peg-out are removed from the available utxos set once the peg-out is created, meaning it is not easy to get them to retrieve the outpoint values for a given segwit peg-out. Storing the outpoint values in an event, when a peg-out is created and before the selected utxos are discarded, is a solution that will allow the PowPeg to retrieve and provide the required outpoint values to the PowHSM to be able to sign a given segwit peg-out.
+
+## Specification
+
+TThe proposal is to create a new event to store the bitcoin transaction hash of the peg-out and the outpoint values of each utxo used as input of the given peg-out. Then, this event will be emitted whenever a transaction spending from the federation address is created. In other words, when a peg-out is created.
```

Comment:
`Then, this event will be emitted whenever a transaction spending from the federation address is created. In other words, when a peg-out is created.`

i would stay with the _transaction spending from the federation address is created_ part.

Reply:
Updated the wording of the paragraph.

Reply:
We may should rewrite the whole paragraph, to just talk about txs being spent from the powpeg.
Because the way it is now, we talk about pegouts first

Reply:
The last sentence states that I'm referring to a peg-out. I expressed it that way just to avoid sounding strange by mentioning each type of peg-out transaction, as Marcos pointed out above.